backends/tfhe-cuda-backend/cuda/src/integer/radix_ciphertext.cu

```diff
@@ -2,8 +2,11 @@
 
 void release_radix_ciphertext_async(cudaStream_t const stream,
                                     uint32_t const gpu_index,
-                                    CudaRadixCiphertextFFI *data) {
-  cuda_drop_async(data->ptr, stream, gpu_index);
+                                    CudaRadixCiphertextFFI *data,
+                                    const bool gpu_memory_allocated) {
+  if (gpu_memory_allocated)
```

Comment:
I think you forgot the brackets to include the frees above this line.

Anyway, we don't need the cuda drop in the conditional.

Reply:
Indeed we don't need this `if`. However we need to free memory on the CPU side as it gets allocated even if `allocate_gpu_memory` is set to false.

Reply:
We'll probably want to come back on this, to have the scratch/cleanup functions stop allocating & freeing memory on the CPU when called with `gpu_memory_allocated=false`, but as mentioned before that would require a lot of extra work and could be done in a second stage.